packages/gatsby-plugin-catch-links/src/catch-links.js

```diff
@@ -78,7 +78,8 @@ module.exports = function(root, cb) {
     // will navigate to an external link instead of doing a pushState resulting
     // in `https://example.com/myapp/https://example.com/not-my-app`
     var re = new RegExp(`^${a2.host}${withPrefix(`/`)}`)
-    if (!re.test(`${a1.host}${a1.pathname}`)) return true
+    var a1pathname = (a1.pathname[0] !== '/' ? '/' : '') + a1.pathname
```

Comment:
can you add comment why this is needed (ie weird quirk)?

Reply:
See https://stackoverflow.com/questions/956233/javascript-pathname-ie-quirk. 

IE does not include the starting slash of an URL in a.pathname. So `a.href="http://example.com/page"` will get `a.pathname="/page"` in chrome et.al (and this is what the standard says) but `a.pathname="page"` in IE.

Reply:
I mean - add comment in code - I got the gist of problem from PR description - but this is for other folks that will read this code later

Reply:
Sorry, of course I will add a comment.

I started looking for the URL API replacement fix but I wasn't sure what this code is really doing. An IE fix for this need to be added to this PR before merging.